### PR TITLE
Make layerEffect adapter agnostic to shadow types.

### PR DESCRIPTION
**Pull request by shaoshing**
This opens the full ability of managing all layer effects. This will allow DS to handle unsupported layer effects, and future supported effects (maybe stroke and fill).

@volfied please review.

### CODE REVIEW

File: src/lib/layerEffect.js [hunk 1]
```diff
@@ -190,128 +194,90 @@ define(function (require, exports) {
...
      * @param {Array.<object>} propertyArray Array of DropShadow properties
      *
      * @return {PlayObject}
      */
-    var _setDropShadows = function (ref, propertyArray) {
-        assert(referenceOf(ref) === "layer", "setDropShadow is passed a non-layer reference");
-
+    var _setLayerEffects = function (ref, effectType, propertyArray) {
+        assert(referenceOf(ref) === "layer", "Adapter._setLayerEffects: ref is a non-layer reference");
+        assert(propertyArray.length > 0, "Adapter._setLayerEffects: propertyArray cannot be emptry");
```
Comment (volfied) [inline src/lib/layerEffect.js]: typo here `emptry`

File: src/lib/layerEffect.js [hunk 2]
```diff
@@ -190,128 +194,90 @@ define(function (require, exports) {
...
         };
     };
+    
     /**
-     * Update multiple drop shadow layer effect properties for the given layer(s)
+     * Update multiple layer effect properties for the given layer(s)
      *
      * @param {ActionDescriptor} ref - Reference of layer(s) to update
+     * @param {string} effectType
      * @param {Array.<object>} propertyArray Array of DropShadow properties
```
Comment (volfied) [inline src/lib/layerEffect.js]: JSDoc needs updating

File: src/lib/layerEffect.js [hunk 3]
```diff
@@ -190,128 +194,90 @@ define(function (require, exports) {
...
-
+    
     /**
-     * Update multiple drop shadow layer effect properties for the given layer(s) without changing the
+     * Update multiple layer effect properties for the given layer(s) without changing the
      * parent layer effect
      *
      * @param {ActionDescriptor} ref - Reference of layer(s) to update
+     * @param {string} effectType
      * @param {Array.<object>} propertyArray Array of DropShadow properties
```
Comment (volfied) [inline src/lib/layerEffect.js]: JSDoc again, search for all `DropShadow`

File: src/lib/layerEffect.js [hunk 4]
```diff
@@ -190,128 +194,90 @@ define(function (require, exports) {
...
      * @return {PlayObject}
      */
-    var setLayerEffect = function (type, ref, propertyArray) {
-        if (type === "innerShadow") {
-            return _setInnerShadows(ref, propertyArray);
-        } else if (type === "dropShadow") {
-            return _setDropShadows(ref, propertyArray);
-        }
+    var setLayerEffect = function (effectType, ref, propertyArray) {
+        return _setLayerEffects(ref, effectType, propertyArray);
```
Comment (volfied) [inline src/lib/layerEffect.js]: We no longer need this wrapper here, so it should be ok to put contents of _setLayerEffects in here, and use this function.

File: src/lib/layerEffect.js [hunk 5]
```diff
@@ -324,26 +290,22 @@ define(function (require, exports) {
...
             // when deleting all effects, keep a hidden effect in Photoshop so that it won't break.
             propertyArray.push(HIDDEN_LAYER_EFFECT_PROPERTIES);
         }
 
-        if (type === "innerShadow") {
-            return _setExtendedInnerShadows(ref, propertyArray);
-        } else if (type === "dropShadow") {
-            return _setExtendedDropShadows(ref, propertyArray);
-        }
+        return _setExtendedLayerEffects(ref, effectType, propertyArray);
```
Comment (volfied) [inline src/lib/layerEffect.js]: Ditto, let's clear out these wrappers completely.

**Comment by volfied** [comment]
Looks good, but needs some clean up. @shaoshing back to you

**Comment by shaoshing** [comment]
Hi @volfied , just as suggested, I fixed the doc and cleaned up the code. Please review again. Thank you.

**Pull request merged**